api/src/main/java/org/openmrs/module/xdssender/api/notificationspullpoint/impl/NotificationsPullPointClientImpl.java

```diff
@@ -58,11 +64,32 @@ public class NotificationsPullPointClientImpl extends WebServiceGatewaySupport i
 	public static final String FACILITY_QNAME = "facility";
 	private static final BigInteger MAX_MESSAGES_PER_REQUEST = BigInteger.valueOf(100);
 
+	private String lastRequestDate = "";
+
 	@Autowired
 	private XdsSenderConfig config;
 
+	@Autowired
+	private CcdService ccdService;
+
 	@Override
 	public List<Message> getNewMessages() {
+		SimpleDateFormat dateFormatter = new SimpleDateFormat(XdsSenderConstants.SOAP_REQUEST_DATE_TIME_FORMAT);
+		TimeZone timeZone = TimeZone.getDefault();
+		dateFormatter.setTimeZone(timeZone);
+		Date newDate = new Date();
+
+		Calendar calendar = Calendar.getInstance();
+		calendar.setTime(newDate);
+		calendar.add(Calendar.DAY_OF_YEAR, -5);
+		Date fiveDaysAgo = calendar.getTime();
+
+		RequestDate lastRequest = ccdService.getLastRequestDate();
+		lastRequestDate = dateFormatter.format(fiveDaysAgo);
+		if (lastRequest != null) {
+			lastRequestDate = dateFormatter.format(lastRequest.getRequestDate());
+		}
```

Comment:
A few things here:

* The 5-days ago calculation is very arbitrary. Why that value? The process runs, by default, once an hour, so 5 days ago is likely to actually be requesting _more_ data than the "give me 100 results" version.
* It's substantially simpler (and more correct) to do this with the `java.time` classes:

```java
String lastRequestDateTime = /* get last date-time timestamp */

if (lastRequestDateTime == null || lastRequestDateTime.isEmpty()) {
   lastRequestDateTime = DateTimeFormatter.ISO_INSTANT.format(Instant.now().minus(2, ChronoUnit.HOURS));
}
```

Then on the storage side, we can just persist the timestamp, e.g.,

```java
if (success) {
   saveLastRun(DateTimeFormatter.ISO_INSTANT.format(Instant.now()))
}
```

Reply:
A technically better solution, though, would be not to use any guessed period and just send a single, final request with the current 100 results, and then start using the `since` from there.